mailbox/api/src/test/java/org/apache/james/mailbox/manager/MailboxManagerFeeder.java

```diff
@@ -1,164 +1,164 @@
+    public static final int USER_COUNT = 3;
+    
+    /**
+     * Number of Sub Mailboxes (mailbox in INBOX) to be created in the Mailbox Manager.
+     */
+    public static final int SUB_MAILBOXES_COUNT = 3;
+    
+    /**
+     * Number of Sub Sub Mailboxes (mailbox in a mailbox under INBOX) to be created in the Mailbox Manager.
+     */
+    public static final int SUB_SUB_MAILBOXES_COUNT = 3;
+    
+    /**
+     * The expected Mailboxes count calculated based on the feeded mails.
+     */
+    public static final int EXPECTED_MAILBOXES_COUNT = DOMAIN_COUNT * 
+                     (USER_COUNT + // INBOX
+                      USER_COUNT * SUB_MAILBOXES_COUNT + // INBOX.SUB_FOLDER
+                      USER_COUNT * SUB_MAILBOXES_COUNT * SUB_SUB_MAILBOXES_COUNT);  // INBOX.SUB_FOLDER.SUBSUB_FOLDER
+    
+    /**
+     * Number of Messages per Mailbox to be created in the Mailbox Manager.
+     */
+    public static final int MESSAGE_PER_MAILBOX_COUNT = 3;
+    
+    /**
+     * Construct a mock mailboxManager based on a valid mailboxManager.
+     * The mailboxManager will be feeded with mailboxes and mails.
+     * 
+     * @param mailboxManager
+     * @throws UnsupportedEncodingException 
+     * @throws MailboxException 
+     */
+    public MailboxManagerFeeder(MailboxManager mailboxManager) throws MailboxException, UnsupportedEncodingException {
+        this.mockMailboxManager = mailboxManager;
+    }
+    
+    /**
+     * @return
+     */
+    public MailboxManager getMockMailboxManager() {
+        return mockMailboxManager;
+    }
+    
+    /**
+     * Utility method to feed the Mailbox Manager with a number of 
+     * mailboxes and messages per mailbox.
+     * 
+     * @throws MailboxException
+     * @throws UnsupportedEncodingException
+     */
+    public void feed() throws MailboxException, UnsupportedEncodingException {
+
+        MailboxPath mailboxPath;
+        
+        for (int i=0; i < DOMAIN_COUNT; i++) {
+
+            for (int j=0; j < USER_COUNT; j++) {
+                
+                String user = "user" + j + "@localhost" + i;
+                
+                String folderName = "INBOX";
+
+                MailboxSession mailboxSession = getMockMailboxManager().createSystemSession(user);
+                mailboxPath = MailboxPath.forUser(user, folderName);
+                createMailbox(mailboxSession, mailboxPath);
+                
+                for (int k=0; k < SUB_MAILBOXES_COUNT; k++) {
+                    
+                    String subFolderName = folderName + ".SUB_FOLDER_" + k;
+                    mailboxPath = MailboxPath.forUser(user, subFolderName);
+                    createMailbox(mailboxSession, mailboxPath);
+                    
+                    for (int l=0; l < SUB_SUB_MAILBOXES_COUNT; l++) {
+
+                        String subSubfolderName = subFolderName + ".SUBSUB_FOLDER_" + l;
+                        mailboxPath = MailboxPath.forUser(user, subSubfolderName);
```

Comment:
no need for variable reutilisation here

Reply:
It's just a class rename. Should I refactor it?

Reply:
Sorry but renaming a "mock"  thing to a "feeder" thing can't come without any code change, a lot of assumption become wrong